pandas/_libs/window.pyx

```diff
@@ -1391,7 +1392,19 @@ def roll_quantile(ndarray[float64_t, cast=True] input, int64_t win,
 
         if nobs >= minp:
             idx = int(quantile * <double>(nobs - 1))
-            output[i] = skiplist.get(idx)
+
+            # Exactly last point
+            if idx == nobs - 1:
+                output[i] = skiplist.get(idx)
+
+            # Interpolated quantile
+            else:
+                qlow = (<double> idx) / (<double>(nobs - 1))
```

Comment:
Bad things (TM) happen if `nobs == 1`, not only because of the obvious division by zero here, but `vhigh` will index the skiplist out of bounds. Unless `minp` can be guaranteed to be at `>= 2` upstream (I don't think it can!) we are going to have to guard against that possibility in the inner loop.

We could simply branch on `nobs >= 1`, which is probably more predictable and faster in the more common cases, or another possibility is to only compute `vhigh` and interpolate if `quantile * (nobs - 1) != idx`.

However this is fixed, a test with enough NaNs to have a single observation in some window, and `min_periods=1`,  should certainly be added.